addons/account/models/account_payment.py

```diff
@@ -212,7 +214,11 @@ def _onchange_partner_id(self):
                 self.partner_bank_account_id = self.partner_id.commercial_partner_id.bank_ids[0]
             else:
                 self.partner_bank_account_id = False
-        return {'domain': {'partner_bank_account_id': [('partner_id', 'in', [self.partner_id.id, self.partner_id.commercial_partner_id.id])]}}
+        if self.payment_type == 'inbound' and self.company_id:
```

Comment:
here you might want to use the company defined on the invoice
```
self.invoice_ids and self.invoice_ids[0].company_id
```

Reply:
Just `self.invoice_ids` is enough as `company_id` is required (from related field) on an invoice.